modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/index/RocksDbSortedIndexStorage.java

```diff
@@ -202,14 +208,47 @@ public ByteBuffer next() {
 
                 this.hasNext = null;
 
-                return ByteBuffer.wrap(key).order(ORDER);
+                return mapper.apply(ByteBuffer.wrap(key).order(ORDER));
+            }
+
+            @Override
+            public @Nullable T peek() {
+                if (hasNext != null && !hasNext) {
+                    return null;
+                }
+
+                T next = null;
+
+                refreshAndPrepareRocksIterator();
+
+                if (!it.isValid()) {
+                    RocksUtils.checkIterator(it);
+                } else {
+                    next = mapper.apply(ByteBuffer.wrap(it.key()).order(ORDER));
+                }
+
+                return next;
             }
 
             private void advanceIfNeeded() throws StorageException {
                 if (hasNext != null) {
                     return;
                 }
 
+                refreshAndPrepareRocksIterator();
```

Comment:
We will now refresh the iterator on every `next` call as well, is that intended?

Reply:
Yep